tests/test_mongodb_views.py

```diff
@@ -0,0 +1,160 @@
+            client["simple_db"]["questions"].insert_many(generate_simple_coll_questions(20))
+
+            # answers has 30 documents
+            client["simple_db"]["answers"].insert_many(generate_simple_coll_answers(30))
+
+            # create view on questions
+            client["simple_db"].command(bson.son.SON([("create", "question_view"), ("viewOn", "questions"), ("pipeline", [])]))
+
+            # create a view by combining two collections
+            client["simple_db"].create_collection(
+                'combined_view',
+                viewOn='questions',
+                pipeline=[{
+                    '$lookup': {
+                        'from': 'answers',
+                        'localField': 'question_id',
+                        'foreignField': 'answer_id',
+                        'as': 'combined_view_final'
+                        }
+                    }]
+                )
+
+    def name(self):
+        return "tap_tester_mongodb_views"
+
+    def tap_name(self):
+        return "tap-mongodb"
+
+    def get_type(self):
+        return "platform.mongodb"
+
+    def get_credentials(self):
+        return {'password': os.getenv('TAP_MONGODB_PASSWORD')}
+
+    def get_properties(self):
+        return {'host': os.getenv('TAP_MONGODB_HOST'),
+                'port': os.getenv('TAP_MONGODB_PORT'),
+                'user': os.getenv('TAP_MONGODB_USER'),
+                'database': os.getenv('TAP_MONGODB_DBNAME'),
+                'include_schemas_in_destination_stream_name': 'true'
+                }
+
+    def expected_check_streams(self):
+        return {'simple_db-questions',
+                'simple_db-answers'}
+
+    def expected_pks(self):
+        return {
+            'simple_db_questions': {'_id'},
+            'simple_db_answers': {'_id'}
+        }
+
+    def expected_row_counts(self):
+        return {
+            'simple_db_questions': 20,
+            'simple_db_answers': 30
+        }
+
+    def expected_sync_streams(self):
+        return {
+            'simple_db_questions',
+            'simple_db_answers'
+        }
+
+    def test_run(self):
+
+        conn_id = connections.ensure_connection(self)
+
+        #  -------------------------------
+        # -----------  Discovery ----------
+        #  -------------------------------
+
+        # run in discovery mode
+        check_job_name = runner.run_check_mode(self, conn_id)
+
+        # verify check exit codes
+        exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+        # verify the tap discovered the right streams
+        found_catalogs = menagerie.get_catalogs(conn_id)
+
+        # validate that the views are not discovered by the tap
+        discovered_streams = [catalog['tap_stream_id'] for catalog in found_catalogs]
+        for stream in discovered_streams:
+            self.assertIn(stream, self.expected_check_streams())
```

Comment:
I think an equality assertion would be better. This would still pass even if one of the streams in `expected_check_streams` was not discovered.

Reply:
done